client/rest-high-level/src/main/java/org/elasticsearch/client/ml/StartDatafeedResponse.java

```diff
@@ -79,8 +79,7 @@ public boolean equals(Object other) {
         }
 
         StartDatafeedResponse that = (StartDatafeedResponse) other;
-        return started == started
-            && Objects.equals(node, that.node);
+        return started && Objects.equals(node, that.node);
```

Comment:
I think the mistake was that we should be comparing to the other object instead of itself.

```suggestion
        return started == that.started && Objects.equals(node, that.node);
```

Reply:
You're definitely right